### PR TITLE
feat(dapps) make wallet connect request data human readable

### PR DESCRIPTION
**Pull request by stefandunca**
### Closes: #15192

- [x] rebased on top of https://github.com/status-im/status-desktop/pull/15643

Notify user if he doesn't hold enough funds to make the transaction
Also check fees funds including the amount to be sent

</img>

### CODE REVIEW

**Comment by stefandunca** [comment]
Beware that this is rebased now on master and require this fix to work https://github.com/status-im/status-desktop/pull/15695

File: ui/StatusQ/src/StatusQ/Core/Utils/AmountsArithmetic.qml [hunk 1]
```diff
@@ -140,6 +140,16 @@ QtObject {
         return amount1.plus(amount2)
     }
 
+    /*!
+      \qmlmethod AmountsArithmetic::sub(amount1, amount2)
+      \brief Returns a Big number whose value is the subtraction of amount2 from amount1.
+     */
+    function sub(amount1, amount2) {
+        console.assert(amount1 instanceof Big.Big)
+        console.assert(amount2 instanceof Big.Big)
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Core/Utils/AmountsArithmetic.qml:149]: Technically, the `amount2` doesn't have to be a `Big`, you can subtract a regular JS Number too

Comment (stefandunca) [reply ui/StatusQ/src/StatusQ/Core/Utils/AmountsArithmetic.qml:149]: Thanks for the heads up, I fixed it like this `console.assert(amount2 instanceof Big.Big || Number.isInteger(amount2))`

File: ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml [hunk 2]
```diff
@@ -436,7 +440,7 @@ SQUtils.QObject {
             let feesInfo = getEstimatedMaxFees(data, method, chainId, mainNetChainId)
 
             let totalMaxFees = Math.sum(feesInfo.maxFees, feesInfo.l1GasFee)
-            let maxFeesEth = Math.div(totalMaxFees, Math.fromString("1000000000"))
+            let maxFeesEth = Math.div(totalMaxFees, Math.fromNumber(1, 9))
```
Comment (Seitseman) [inline ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml:443]: Maybe using name `Math` for `SQUtils.AmountArithmetic` was not the best idea, as it usually denotes js static object

File: ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml [hunk 3]
```diff
@@ -503,6 +526,87 @@ SQUtils.QObject {
...
+                    const stringPayload = SessionRequest.methods.signTypedData_v4.getMessageFromData(data)
+                    payload = JSON.stringify(JSON.parse(stringPayload), null, 2)
+                    break
+                }
+                case SessionRequest.methods.signTypedData.name: {
+                    const stringPayload = SessionRequest.methods.signTypedData.getMessageFromData(data)
+                    payload = JSON.stringify(JSON.parse(stringPayload), null, 2)
+                    break
+                }
+                default:
```
Comment (Seitseman) [inline ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml:555]: Would it makes sense to case/break SessionRequest.methods sendTransaction and in default case log an error or warning message?

**Review by Seitseman — COMMENTED** [review]
Looks good!

File: ui/StatusQ/src/StatusQ/Core/Utils/AmountsArithmetic.qml [hunk 4]
```diff
@@ -140,6 +140,16 @@ QtObject {
         return amount1.plus(amount2)
     }
 
+    /*!
+      \qmlmethod AmountsArithmetic::sub(amount1, amount2)
+      \brief Returns a Big number whose value is the subtraction of amount2 from amount1.
+     */
+    function sub(amount1, amount2) {
+        console.assert(amount1 instanceof Big.Big)
+        console.assert(amount2 instanceof Big.Big || Number.isInteger(amount2))
```
Comment (Seitseman) [inline ui/StatusQ/src/StatusQ/Core/Utils/AmountsArithmetic.qml:149]: https://mikemcl.github.io/big.js/#minus allows string as well, is it expected for us to also allow strings here?

**Review by Seitseman — APPROVED** [review]

**Comment by stefandunca** [comment]
@Seitseman I merge it as it is for now but will follow up with your recommendations in the next PR fix.

File: ui/StatusQ/src/StatusQ/Core/Utils/AmountsArithmetic.qml (hunk 4 above)
Comment (stefandunca) [reply ui/StatusQ/src/StatusQ/Core/Utils/AmountsArithmetic.qml:149]: I don't think we should be too flexible. Better explicitly call from string instead.

**Pull request merged**